worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala

```diff
@@ -742,7 +742,12 @@ private[celeborn] class Worker(
     synchronized {
       val expiredApplicationIds = new JHashSet[String]()
       expiredShuffleKeys.asScala.foreach { shuffleKey =>
-        partitionLocationInfo.removeShuffle(shuffleKey)
+        partitionLocationInfo.removeShuffle(shuffleKey).foreach { loc =>
```

Comment:
StorageManager will remove expired shuffle keys which will destroy the writerd and return buffer.